controllers/teams.js

```diff
@@ -0,0 +1,201 @@
+            if ( player.isReady )
+                return player;
+        });
+        if ( readyPlayers.length )
+            return res.send( readyPlayers );
+        return res.send( 'Non of the players is ready' );
+    });
+}
+
+// TODO define route for users of team
+exports.getUsersOfTeam = ( req, res, next ) => {
+    Team.findById( req.body.teamId, ( err, team ) => {
+        if ( err ) return next( err );
+        if ( !team ) return res.status( 404 ).send( 'No team with that ID' );
+        return res.status( 200 ).send( team.players );
+    });
+}
+
+exports.toggleReadyUser = ( req, res, next ) => {
+    if ( !req.body.playerId )
+        return res.status( 400 ).send( 'Missing player ID' );
+
+	Team.findById( req.params.id || req.body.userTeam, ( err, team ) => {
+        if ( err ) return next( err );
+        if ( !team ) return res.status( 404 ).send( 'No team with that ID' );
+
+        for ( var i in team.players ) {
+            if ( team.players[ i ].playerId.toString() === req.body.playerId.toString() ) {
+                console.log( 'about to toggle' )
+                team.players[ i ].isReady = !team.players[ i ].isReady;
+                console.log( 'player\'s isReady is toggled' );
+                var status = team.players[ i ].isReady;
+                break;
+            }
+        }
+        team.markModified( 'players' );
+        team.save( ( err ) => {
+            if ( err ) return next( err );
+            console.log( 'player: ' + req.body.playerId + ' updated to: ' + status );
+            return res.sendStatus( 200 );
+        });
+    });
+}
+
+exports.addUserToTeam = ( req, res, next ) => {
+    if ( !req.body.playerId ) return res.status( 400 ).send( 'no player ID' );
+
+    Promise.all([
+        User.findById( req.body.playerId ).exec(),
+        Team.findById( req.body.teamId ).exec()
+
+    ]).then( ( results ) => {
+        let user = results[ 0 ];    // get user
+        let team = results[ 1 ];    // get team
+
+        if ( !team )
+            return res.status( 404 ).send( 'No team with that ID' );
+        if ( team.players.length === 5 ) {
+            var err = new Error( 'Max number of players is 5' );
+            err.status = 400;
+            throw err;
+        }
+
+        // check if player is already member of the team
+        for ( var i in team.players ) {
+            if ( team.players[ i ].playerId == req.body.playerId ) {
+                var err = new Error( 'This player already member of your team' );
+                err.status = 400;
+                throw err;
+            }
+        }
+
+        // add player to the team
+        team.players.push({
+            playerId: req.body.playerId
+        });
+
+        team.markModified( 'players');
+        team.save();
+
+        // add team to player
+        user.team = req.body.teamId;
+        user.markModified( 'team' );
+        user.save();
+
+    }).then( () => {
+        return res.sendStatus( 200 );

```

Comment:
Shouldn't this be `201` ?

Reply:
for both add/remove user, or `201` for add, `200` for remove?

Reply:
Correct, 201 is for adding any resource. 200 means action succeeded, so it is a generic one.